data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/ScanObjectWorker.java

```diff
@@ -209,6 +215,17 @@ private void startProcessingObject(final long waitTimeMillis) {
                     }
                     partitionKeys.remove(objectToProcess.get().getPartitionKey());
                 }, ACKNOWLEDGEMENT_SET_TIMEOUT);
+
+                acknowledgementSet.addProgressCheck(
```

Comment:
When will this scheduled event gets unregistered? 
Also, do we register another similar scheduled event for DynamoDB pull based source?

Reply:
Looking at the code for AcknowledgmentSet, the acknowledgmentSet.complete() call cancels the future for this progress check